routes/users.js

```diff
@@ -17,6 +17,7 @@ router.patch("/self", authenticate, userValidator.updateUser, users.updateSelf);
 router.get("/", userValidator.getUsers, users.getUsers);
 router.get("/self", authenticate, users.getSelfDetails);
 router.get("/isUsernameAvailable/:username", authenticate, users.getUsernameAvailabilty);
+router.get("/username", authenticate, userValidator.getUsername, users.getUsername);
```

Comment:
Not a requirement now but would suggest - `/username-generation` for future

Reply:
Okk @pallabez Thank you